### PR TITLE
Flask: update GET /api/users, move to users.py, add tests

### PR DESCRIPTION
**Pull request by kevinlul**
Now returns an array of JSON objects of the username, email, admin status, last login, activation status, and an array of group codes.
Also hook up the user list to use this updated endpoint.

#193

### CODE REVIEW

File: flask/tests/test_users.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+def test_list_users(test_database, client, login_as):
+    login_as("admin")
+    response = client.get("/api/users")
+    assert response.status_code == 200
+
+    user_list = response.get_json()
+    assert len(user_list) == 2
+
+    # return order is unspecified
+    user_list.sort(key=lambda user: user["username"])
```
Comment (SamarthVP) [inline flask/tests/test_users.py:13]: Clever! Out of curiosity, does SQLAlchemy by default obey some kind of ordering or are collections returned in no particular order.

**Review by SamarthVP — APPROVED** [review]
Solid backend changes

File: flask/tests/test_users.py (hunk 1 above)
Comment (kevinlul) [reply flask/tests/test_users.py:13]: SQLAlchemy should be returning in the same order as the underlying SQL, which is unspecified unless an ORDER BY clause is used. So the default ordering is completely arbitrary and probably the order the database has in its index, which is likely insertion order.

**Review by frewmack — APPROVED** [review]
Frontend looks good to me!

**Pull request merged**